### PR TITLE
Added logic for the public feed

### PR DESCRIPTION
**Pull request by Open-Fuse**
Created the MessageFeed class

Added the method getAllMessages() to the Datastore class.

Refactored getMessages() and getAllMessages() by extracting and calling the helper method processMessageQuery()

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/MessageFeedServlet.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  
+ private Datastore datastore;
+
+ @Override
+ public void init() {
+  datastore = new Datastore();
+ }
+ 
+ /**
+  * Responds with a JSON representation of Message data for all users.
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/servlets/MessageFeedServlet.java:29]: Please add an example of the JSON returned in this comment. Something like`Example JSON: {...however it looks...}`

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 2]
```diff
@@ -54,33 +54,60 @@ public void storeMessage(Message message) {
...
+	  
+	  List<Message> messages = this.processMessageQuery(results);
+
+	  return messages;
+ }
+  
+  /**
+   * Iterates through query results and returns a list of messages
+   * 
+   * @return a a list of messages.
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/data/Datastore.java]: Two `a`s here :)

File: src/main/java/com/google/codeu/data/Datastore.java [hunk 3]
```diff
@@ -54,33 +54,60 @@ public void storeMessage(Message message) {
...
+   */
+  private List<Message> processMessageQuery(PreparedQuery queryResults) {
```
Comment (NickChapman) [inline src/main/java/com/google/codeu/data/Datastore.java:86]: This is a great refactor! Very nice that it doesn't repeat the logic in both places 👍

**Review by NickChapman — CHANGES_REQUESTED** [review]

File: src/main/webapp/js/feed-loader.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+
+// Fetch messages and add them to the page.
+function fetchMessages() {
+    const url = '/feed';
+    fetch(url).then((response) => {
+        console.log(response);
```
Comment (NickChapman) [inline src/main/webapp/js/feed-loader.js]: Do you want to keep this `console.log` or what it just for debugging purposes?

**Review by NickChapman — COMMENTED** [review]
Do you have an example screenshot of how the message feed looks?

File: src/main/webapp/js/feed-loader.js (hunk 4 above)
Comment (Open-Fuse) [reply src/main/webapp/js/feed-loader.js]: This is for debugging, I'll remove it. Plus, where should I post the screenshot?

**Review by NickChapman — APPROVED** [review]
Looks good, let's get this submitted!

**Pull request merged**